### PR TITLE
Limit gathered phrase indexes to excerpt fields

### LINKED ISSUE #1607: Only include phrase indexes for fields that we need excerpts for
The phrase indexes that are gather could be reduced to only those fields that we are actually gathering excerpts for.

### PR DESCRIPTION
**Pull request by lbschanno**
Reduce the phrase indexes gathered to only the fields that we need to
gather excerpts for.

Resolves #1607

### CODE REVIEW

File: warehouse/query-core/src/test/java/datawave/query/jexl/functions/ContentFunctionsTest.java [hunk 1]
```diff
@@ -1825,4 +1948,52 @@ public void testIgnoreIrrelevantZones() {
...
+        
+        Assert.assertTrue(expect(o, true));
+        assertPhraseOffsetsEmpty();
+    }
+    
+    /**
+     * Verify that if gatherPhraseOffsets is true, if a phrase index for is found for a non-excerpt field, it is not recorded.
+     */
+    @Test
+    public void testNonMatchingExcerptFields() {
```
Comment (apmoriarty) [inline warehouse/query-core/src/test/java/datawave/query/jexl/functions/ContentFunctionsTest.java:1973]: An alternate test for this would be an event with a phase for two different fields (BODY and CONTENT), excerpts enabled, and excerpt fields set to only one field. Not required for approval, just another edge case to consider.

**Review by apmoriarty — DISMISSED** [review]

**Review by ivakegg — DISMISSED** [review]

File: warehouse/query-core/src/test/java/datawave/query/jexl/functions/ContentFunctionsTest.java (hunk 1 above)
Comment (lbschanno) [reply warehouse/query-core/src/test/java/datawave/query/jexl/functions/ContentFunctionsTest.java:1973]: This was a long time coming, but I added a test to cover this edge case.

**Comment by ivakegg** [comment]
Some regex forms that I am seeing:
1111.*
1111.*?
1111\d*
.*?1111
.*1111
.*1111.*
^11[23].*
1111[0-9]{5}
.*1111\..*

**Review by ivakegg — DISMISSED** [review]

File: warehouse/query-core/src/test/java/datawave/query/jexl/functions/ContentFunctionsTest.java (hunk 1 above)
Comment (apmoriarty) [reply warehouse/query-core/src/test/java/datawave/query/jexl/functions/ContentFunctionsTest.java:1973]: Perfect, thank you! Apologies for the lapse in attention

**Review by apmoriarty — DISMISSED** [review]

**Comment by apmoriarty** [comment]
I would really like to get this change in soon, it's been lingering for far too long. Resolved merge conflicts and pushed those updates.

**Review by ivakegg — DISMISSED** [review]

**Review by ivakegg — APPROVED** [review]
